### PR TITLE
test: remove LOG and LOGF variadic macros

### PR DESCRIPTION
**Pull request by saghul**
Initial patch by @simar7, thanks!

### CODE REVIEW

**Comment by saghul** [comment]
R=@bnoordhuis

This supesedes #185, hopefully I addressed all issues pointed out there.

**Comment by simar7** [comment]
Sweet, thanks @saghul

**Comment by bnoordhuis** [comment]
Rubber-stamp LGTM.  Can I suggest you check the CI before merging?

**Comment by saghul** [comment]
I did, but then added a couple of more fixes. I'll run it again before
merging.

Thanks!

File: test/runner.c [hunk 1]
```diff
@@ -320,49 +323,59 @@ int run_test(const char* test,
...
+      fflush(stderr);
     } else if (status == TEST_SKIP) {
-      LOGF("\n`%s` skipped\n", test);
+      fprintf(stderr, "\n`%s` skipped\n", test);
+      fflush(stderr);
     } else if (status != 0) {
-      LOGF("\n`%s` failed: %s\n", test, errmsg);
+      fprintf(stderr, "\n`%s` failed: %s\n", test, errmsg);
+      fflush(stderr);
     } else {
```
Comment (Pyrohh) [inline test/runner.c]: Instead of one inside each, why not just have one `fflush(stderr)` after the if statements?

**Comment by saghul** [comment]
CI run: https://jenkins-iojs.nodesource.com/view/libuv/job/libuv+any-pr+multi/83/

File: test/runner.c (hunk 1 above)
Comment (saghul) [inline test/runner.c]: Good catch!

**Pull request merged**